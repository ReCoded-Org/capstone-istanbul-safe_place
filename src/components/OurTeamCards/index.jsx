import React from "react";
import { Container, Row } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import gitHubIcon from "../../images/icons/githubLogo.svg";
import linkedInIcon from "../../images/icons/linkedInLogo.svg";
import defaultImg from "../../images/icons/defaultImg.svg";
import louisImg from "../../images/louisLiSmiling.jpeg";
import ammarImg from "../../images/ammarSammourSmiling.jpeg";
import jennetImg from "../../images/jennetHydyrovaSmiling.jpg";
import hibaImg from "../../images/hibaMachfejSmiling.png";
import jamilImg from "../../images/jamilBailonySmiling.png";
import mohammadImg from "../../images/mohammadJundiSmiling.jpg";
import "./index.scss";

const OurTeamCards = () => {
  const { t } = useTranslation();
  const membersInfoArr = [
    ...t("aboutUsPage.ourTeamInfo", { returnObjects: true }),
  ];
  const memberImagesMap = {
    louisImg: louisImg,
    ammarImg: ammarImg,
    jennetImg: jennetImg,
    hibaImg: hibaImg,
    jamilImg: jamilImg,
    mohammadImg: mohammadImg,
  };

  return (
    <Container className="ourTeamInfo">
      <h4>{t("aboutUsPage.subHeader")}</h4>
      <Row className="justify-content-md-center">
        {membersInfoArr.map((member) => {
          return (
            <div className="singleMemberInfo" key={member.name}>
              <img
                src={memberImagesMap[member.portraitImg] || defaultImg}
                alt={member.name}
                className="portraitImg"
              />
              <div>
                {member.gitHubUrl && (
                  <a
                    href={member.gitHubUrl}
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    <img src={gitHubIcon} alt="GitHub icon" />
                  </a>
                )}
                {member.linkedInUrl && (
                  <a
                    href={member.linkedInUrl}
                    target="_blank"
                    rel="noopener noreferrer"
                    className="linkedInIcon"
                  >
                    <img src={linkedInIcon} alt="LinkedIn icon" />
                  </a>
                )}
              </div>
              <h5>{member.name}</h5>
              <h6>{member.title}</h6>
            </div>
          );
        })}
      </Row>
    </Container>
  );
};

export default OurTeamCards;
