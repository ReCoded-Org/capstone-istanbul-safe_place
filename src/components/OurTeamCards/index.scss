@import "../../style/variables";

.ourTeamInfo {
  .linkedInIcon {
    margin-left: 0.5rem;
  }
  .portraitImg {
    height: 13rem;
    width: 13rem;
    border-radius: 10rem;
    border: 0.22rem solid;
    border-color: $primaryColor;
    background-color: $primaryColor;
  }
  .singleMemberInfo {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 1rem 0.8rem;
  }
  h4 {
    color: $successColor;
    font-weight: bold;
    font-size: 1.5rem;
    font-family: $mainFontFamily;
    text-align: center;
    margin-top: 3rem;
  }
  h5 {
    margin-top: 0.7rem;
    font-size: 1.25rem;
    color: $primaryColor;
  }
  h6 {
    font-size: 0.875rem;
    color: $primaryColor;
  }
}
