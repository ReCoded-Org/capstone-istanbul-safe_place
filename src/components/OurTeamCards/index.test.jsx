import React from "react";
import OurTeamCards from "./index.jsx";
import renderer from "react-test-renderer";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

it("renders correctly and matches the snapshot", () => {
  const tree = renderer.create(<OurTeamCards />).toJSON();
  expect(tree).toMatchSnapshot();
});
