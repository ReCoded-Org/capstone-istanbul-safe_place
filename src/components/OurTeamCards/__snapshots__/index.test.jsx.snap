// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div
  className="ourTeamInfo container"
>
  <h4>
    aboutUsPage.subHeader
  </h4>
  <div
    className="justify-content-center row"
  >
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
    <div
      className="singleMemberInfo"
    >
      <img
        className="portraitImg"
        src="defaultImg.svg"
      />
      <div />
      <h5 />
      <h6 />
    </div>
  </div>
</div>
`;
